benchmarks/benchmark-dependencies-methods-polars.py

```diff
@@ -0,0 +1,385 @@
+        df["type"] = df["type"].astype("int32[pyarrow]")
+        df["version"] = df["version"].astype("string[pyarrow]")
+        df.index = df.index.astype(audb.core.define.DEPEND_INDEX_DTYPE)
+        # Set dtypes in library
+        audb.core.define.DEPEND_FIELD_DTYPES = {
+            "archive": "string[pyarrow]",
+            "bit_depth": "int32[pyarrow]",
+            "channels": "int32[pyarrow]",
+            "checksum": "string[pyarrow]",
+            "duration": "float64[pyarrow]",
+            "format": "string[pyarrow]",
+            "removed": "int32[pyarrow]",
+            "sampling_rate": "int32[pyarrow]",
+            "type": "int32[pyarrow]",
+            "version": "string[pyarrow]",
+        }
+    return df
+
+
+# === Dependencies pandas.DataFrame ===
+data_cache = audeer.path(cache, f"df.{CACHE_EXT}")
+
+num_rows = 1000000
+
+if not os.path.exists(data_cache):
+    bit_depths = [0, 16, 24]
+    channels = [0, 1, 2]
+    formats = ["csv", "wav", "txt"]
+    sampling_rates = [0, 16000, 44100]
+    types = [0, 1, 2]
+    versions = ["1.0.0", "1.1.0"]
+    records = [
+        {
+            "file": f"file-{n}.wav",
+            "archive": f"archive-{n}",
+            "bit_depth": random.choices(bit_depths, weights=[0.1, 0.8, 0.1])[0],
+            "channels": random.choices(channels, weights=[0.1, 0.8, 0.1])[0],
+            "checksum": hashlib.md5(
+                pickle.dumps(random.choice(string.ascii_letters))
+            ).hexdigest(),
+            "duration": 10 * random.random(),
+            "format": random.choices(formats, weights=[0.1, 0.8, 0.1])[0],
+            "removed": random.choices([0, 1], weights=[0.1, 0.9])[0],
+            "sampling_rate": random.choices(sampling_rates, weights=[0.1, 0.8, 0.1])[0],
+            "type": random.choices(types, weights=[0.1, 0.8, 0.1])[0],
+            "version": random.choices(versions, weights=[0.2, 0.8])[0],
+        }
+        for n in range(num_rows)
+    ]
+    df = pd.DataFrame.from_records(records)
+    for column, dtype in zip(
+        audb.core.define.DEPEND_FIELD_NAMES.values(),
+        audb.core.define.DEPEND_FIELD_DTYPES.values(),
+    ):
+        df[column] = df[column].astype(dtype)
+    df.set_index("file", inplace=True)
+    df.index.name = None
+
+    df.index = df.index.astype(audb.core.define.DEPEND_INDEX_DTYPE)
+    df.to_pickle(data_cache)
+
+    # df.index = df.index.astype("string")
+    if CACHE_EXT == "pkl":
+        df.to_pickle(data_cache)
+    elif CACHE_EXT == "parquet":
+        df.to_parquet(**PARQUET_SAVE_OPTS)
+
+
+# ===== Benchmark audb.Dependencies =====
+# deps = audb.Dependencies()
+# deps.load(data_cache)
+file = "file-10.wav"
+n_files = 10000
+results = pd.DataFrame(columns=dtypes)
+results.index.name = "method"
+set_dependency_module()
+
+for dtype in dtypes:
+    # load them
+    deps = audb.Dependencies()
+    deps.load(data_cache)
+    deps._df = astype(deps._df, dtype)
```

Comment:
This sets all dtypes to "string", shouldn't be an integer dtype faster for integers?

Reply:
OK, I see the `nan` in the benchmark results. So you tested only for string types, correct?

Reply:
I have changed all this slightly.  Polars does not allow you to create a series of type object, so there are only strings.  The type definitions are in `utils.DEPEND_FIELD_DTYPES` .   

`__init__` then creates the empty df. And as polars has a scheme this is used from there onwards.